### PR TITLE
add Retry feature and --max-retry option to `mkr throw`

### PR DESCRIPTION
**Pull request by astj**
- Retry with exponential backoff (min interval is 15 sec and Max is 5 min).
- Don't retry when HTTP response is 4xx

demo:
```
[github.com/mackerelio/mkr]$ echo "tcp.CLOSING 0 1397031808" | ./mkr --apibase http://localhost:5000/ --conf /usr/local/etc/mackerel-agent.conf throw --host shiba
     error API request failed: 404 Not Found
[github.com/mackerelio/mkr]$ echo "tcp.CLOSING 0 1397031808" | ./mkr --apibase http://localhost:5000/ --conf /usr/local/etc/mackerel-agent.conf throw --host shiba
     error API request failed: 500 Internal Server Error
[github.com/mackerelio/mkr]$ echo "tcp.CLOSING 0 1397031808" | ./mkr --apibase http://localhost:5000/ --conf /usr/local/etc/mackerel-agent.conf throw --host shiba --max-retry 2
   warning Failed to request. will retry after 15 seconds. Error: API request failed: 500 Internal Server Error
   warning Failed to request. will retry after 30 seconds. Error: API request failed: 500 Internal Server Error
     error API request failed: 500 Internal Server Error
[github.com/mackerelio/mkr]$ echo "tcp.CLOSING 0 1397031808" | ./mkr --apibase http://localhost:5000/ --conf /usr/local/etc/mackerel-agent.conf throw --host shiba --max-retry 1
     error API request failed: 403 Forbidden
[github.com/mackerelio/mkr]$
```

### CODE REVIEW

File: throw.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    Post metric values to 'host metric' or 'service metric'.
+    Output format of metric values are compatible with that of a Sensu plugin.
+    Requests "POST /api/v0/tsdb". See https://mackerel.io/api-docs/entry/host-metrics#post .
+    Automatically retries the API request when --max-retry is specified.
+`,
+	Action: doThrow,
+	Flags: []cli.Flag{
+		cli.StringFlag{Name: "host, H", Value: "", Usage: "Post host metric values to <hostID>."},
+		cli.StringFlag{Name: "service, s", Value: "", Usage: "Post service metric values to <service>."},
+		cli.IntFlag{Name: "max-retry, m", Usage: "Retries up to N times when API request fails."},
```
Comment (itchyny) [inline throw.go]: Not a strong opinion but how about `--retry, -r`? I think the short option `-m` is hard to tell what it means.

**Comment by itchyny** [comment]
LGTM!

**Comment by astj** [comment]
Thank you!

**Comment by astj** [comment]
Oops, I should have changed p-r title before merging ...

**Pull request merged**